llvmlite/ir/instructions.py

```diff
@@ -162,7 +162,7 @@ def descr(self, buf):
 class PredictableInstr(Instruction):
 
     def set_weights(self, weights):
-        operands = [MetaDataString(self.module, "branch_weights")]
+        operands = [MetaDataString(self.module, b"branch_weights")]
```

Comment:
Well, the LLVM Language Reference doesn't say anything about the character encoding in which the IR is to be interpreted, so we have to find some compromise.

Arrays contain binary data, this is certain. Accepting text for them is at odds with the Python 3 model of separating text and bytes: if a frontend knows a string must be utf-8 encoded, therefore, it should encode it itself (after all, the string could be utf-16 encoded instead, e.g. for use with the Windows Unicode filesystem APIs).

I would lean towards considering metadata strings are human-readable text, but I suppose there may be cases where they are not. My concern here is that most existing code probably passes a `str` value (i.e. unicode on Python 3), not a `bytes`.